Benjamin - Benjamin-Tu/bt_week4_assignment.do

```diff
@@ -0,0 +1,188 @@
+
+
+global wd "/Users/benjamintu/Downloads/Econ490_Fall2022/Week4/04_assignment/data"
+
+use "$wd/village_pixel.dta", clear
+
+* Q1 a)
+egen pixel_mean = mean(payout),by(pixel) //calculate the mean for each different pixel.
+
+gen pixel_consistent = 0 if pixel_mean== 0 | pixel_mean ==1    // pixel is consistent if the mean is an integer.
+
+replace pixel_consistent = 1 if pixel_mean!= 0 & pixel_mean!= 1 // if the mean is not an integer, pixel is not consistent
+
+* b)
+/*
+why the value changes when I try to destring?
```

Comment:
I'm not totally sure what your question is asking here. Are you wondering why running `mean` and `by` does not produce a dummy variable here? Happy to help here if I understand why you wanted to run these few lines of code (but ran into trouble), what (if any) part of your existing code these lines of code would replace, and how the results differ from what you'd expected.